src/library/scala/concurrent/duration/Duration.scala

```diff
@@ -124,6 +124,8 @@ object Duration {
       Undefined
     else if (nanos > Long.MaxValue || nanos < Long.MinValue)
       throw new IllegalArgumentException("trying to construct too large duration with " + nanos + "ns")
+    else if (nanos < 0)
+      fromNanos((nanos - 0.5).toLong)
```

Comment:
@adriaanm, if `Double` is unstable for big values (`~10^13`), shouldn't we disallow `nanos` as a `Double`, if its sole purpose seems to be to infinite `Duration` (I'm sure `Long.MaxValue` nanos is pretty close to infinite anyway, i.e. `300` years)?